distributed/nanny.py

```diff
@@ -420,31 +419,7 @@ async def instantiate(self) -> Status:
                 pre_spawn_env=self.pre_spawn_env,
                 config=self.config,
             )
-
-        if self.death_timeout:
-            try:
-                result = await asyncio.wait_for(
-                    self.process.start(), self.death_timeout
-                )
-            except asyncio.TimeoutError:
-                logger.error(
-                    "Timed out connecting Nanny '%s' to scheduler '%s'",
-                    self,
-                    self.scheduler_addr,
-                )
-                await self.close(
-                    timeout=self.death_timeout, reason="nanny-instantiate-timeout"
-                )
-                raise
-
-        else:
-            try:
-                result = await self.process.start()
-            except Exception:
-                logger.error("Failed to start process", exc_info=True)
-                await self.close(reason="nanny-instantiate-failed")
-                raise
-        return result
```

Comment:
`Server.start` takes care of closing it if something happens during startup. All of this special handling is unnecessary.
The treatment of death_timeout is also handled in `Server.start` so this PR introduces a much more streamlined approach to the lifecycle management of the Nanny

Reply:
Which reminds me... if there is a startup failure during restart, we're not robust anymore. I'll add an extra test and adjust the behavior